geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -112,9 +112,11 @@ trait KafkaDataStoreSchemaManager extends DataStore with LazyLogging {
     }.getOrElse(None)
   }
 
-  override def getNames: util.List[Name] =
-    zkUtils.zkClient.getChildren(zkPath).asScala.map(name => new NameImpl(name) : Name).asJava
-
+  override def getNames: util.List[Name] = {
+    zkUtils.zkClient.getChildren(zkPath).asScala
+      .filter(name => zkUtils.zkClient.exists(getTopicPath(name)))
```

Comment:
Dead topic names accumulating in zookeeper would depend on the user. removeSchema will remove the zk nodes pertaining to the schema and calling createSchema with a zkpath + sft name that already exists will throw an IllegalArgumentException.